### PR TITLE
[QF-4407]feat(study-modal): add embedded audio player to study mode modal footer 

### PR DESCRIPTION
**Pull request by yousefdergham**
## Summary                             
                                                                                                                            
 Adds an embedded audio player to the study mode modal footer so users can control playback without leaving the modal.      

  **What's included:**
  - Full audio player (progress slider + playback controls) rendered in the modal footer when audio is playing
  - Adapted 3-dots menu for the modal context (repeat and experience options removed)
  - Volume and reciter popovers adjusted for modal layout (open upward, correct z-index, single scroll)
  - Auto-scroll paused on the reading page while the modal is open
  - Main page-level audio player properly restored when the modal is closed
  
  Closes: [QF-4407](https://quranfoundation.atlassian.net/browse/QF-4407)

  ## Type of Change

  - [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
  - [x] ✨ New feature (non-breaking change which adds functionality)
  - [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as
        expected)
  - [ ] 📝 Documentation update
  - [ ] ♻️  Refactoring (no functional changes)

  ## Scope Confirmation

  - [x] This PR addresses **one** feature/fix only
  - [ ] If multiple changes were needed, they are split into separate PRs

  ## Rollback Safety

  - [x] Can be safely reverted without data issues or migrations
  - [ ] Rollback requires special steps (describe below):

  ## Test Plan

  - [ ] Unit tests added/updated
  - [ ] Integration tests added/updated
  - [x] Manual testing performed

  **Testing steps:**

  1. Open a surah page and start audio playback
  2. Open the study modal — verify the audio player appears in the modal footer with slider and controls
  3. Verify play/pause, prev/next, volume, and close all work inside the modal
  4. Verify the progress slider is positioned correctly inside the modal (not at the viewport bottom)
  5. Open the 3-dots menu — verify "Manage Repeat Settings" and "Experience" are hidden, but Download, Speed, and Reciter
  remain
  6. Open the reciter list — verify single scroll (no double scrollbar)
  7. Verify volume popover opens upward and is visible above the modal overlay
  8. Open the verse search dropdown in the header — verify it appears above the audio footer
  9. Verify the page behind the modal does NOT auto-scroll while the modal is open
  10. Close the modal — verify the main page-level audio player reappears and audio keeps playing
  11. Repeat steps 2-10 on an SSR study modal page (e.g. tafsir route)
  12. Close the SSR modal — verify the main audio player reappears on the reading page
  13. Open the modal with no audio playing — verify no footer is shown

  ### Edge Cases Verified

  - [x] ⏳ Loading state handled
  - [x] ❌ Error state handled
  - [x] 📭 Empty state handled (no footer when audio not playing)
  - [ ] 👤 Logged-in vs guest behavior (if applicable)

  ## Pre-Review Checklist

  ### Code Quality

  - [x] I have performed a **self-review** of my code (file by file)
  - [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
  - [x] No `any` types used (or justified if unavoidable)
  - [x] No unused code, imports, or dead code included
  - [x] Complex logic has inline comments explaining "why"
  - [x] Functions are under 30 lines and follow single responsibility

  ### Testing & Validation

  - [ ] All tests pass locally (`yarn test`)
  - [x] Linting passes (`yarn lint`)
  - [x] Build succeeds (`yarn build`)
  - [x] Edge cases and error scenarios are handled

  ### Documentation

  - [x] Code is self-documenting with clear naming
  - [ ] README updated (if adding features or setup changes)
  - [x] Inline comments added for complex logic

  ### Localization (if UI changes)

  - [x] All user-facing text uses `next-translate`
  - [x] Only English locale files modified (Lokalise handles others)
  - [x] RTL layout verified

  ### Accessibility (if UI changes)

  - [x] Semantic HTML elements used
  - [x] ARIA attributes added where needed
  - [x] Keyboard navigation works

  ## AI Assistance Disclosure

  - [ ] AI tools were NOT used for this PR
  - [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4407]: https://quranfoundation.atlassian.net/browse/QF-4407?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/AudioPlayer/AudioPlayer.tsx [hunk 1]
```diff
@@ -133,7 +136,7 @@ const AudioPlayer = () => {
     <>
       <div
         className={classNames(styles.container, styles.containerDefault, {
-          [styles.containerHidden]: !isVisible,
+          [styles.containerHidden]: !isVisible || isStudyModeOpen,
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/AudioPlayer/AudioPlayer.tsx:139]: good catch
